driver-core/src/test/java/com/datastax/driver/core/CCMBridge.java

```diff
@@ -1048,6 +1047,34 @@ public CCMBridge build() {
         cassandraVersion = this.version;
       }
 
+      selectedPorts.clear();
+      // Preemptively banning every number that looks like explicitly provided port
+      for (Map.Entry<String, Object> entry : this.cassandraConfiguration.entrySet()) {
+        Object value = entry.getValue();
+        try {
+          Integer result = Integer.parseInt(value.toString());
+          if ((1 <= result) && (result <= 65535)) {
+            selectedPorts.add(result);
+          }
+        } catch (NumberFormatException e) {
+        }
+      }
+      if (dseVersion != null) {
+        for (Map.Entry<String, Object> entry : this.dseConfiguration.entrySet()) {
+          Object value = entry.getValue();
+          try {
+            Integer result = Integer.parseInt(value.toString());
+            if ((1 <= result) && (result <= 65535)) {
+              selectedPorts.add(result);
+            }
+          } catch (NumberFormatException e) {
```

Comment:
It's not a good idea to just ignore this. Maybe it's better to fail when the port is not actually a number?